app/models/item.rb

```diff
@@ -36,4 +36,10 @@ def self.least_pop_items
   def quantity_ordered
     item_orders.sum(:quantity)
   end
+
+  def discount_price
+    return price if Discount.find_by(merchant_id: self.merchant_id).nil?
+    discount = Discount.find_by(merchant_id: self.merchant_id)
+    price * (1 - discount.percentage_discount.to_f / 100 )
```

Comment:
Rethought the way I was calculating discount price!! Got it all fixed by changing the percentage discount to a decimal.